substratest/client.py

```diff
@@ -150,6 +150,12 @@ def add_compute_plan(self, spec, *args, **kwargs):
         self.state.compute_plans.append(compute_plan)
         return compute_plan
 
+    def update_compute_plan(self, spec, *args, **kwargs):
+        res = self._client.update_compute_plan(spec.compute_plan_id, spec.to_dict(), *args, **kwargs)
+        compute_plan = assets.ComputePlan.load(res).attach(self)
+        self.state.compute_plans.append(compute_plan)
```

Comment:
This shouldn't be an append but rather a replace I think.

Reply:
Good comment.
As discussed previously, the session state is not very usable right now and does not contain the last up-to-date assets.

As a short term fix we should remove the `append()`. We will see later how the state could be improved.

Reply:
Ups, to be consistent with existing code the best solution seems to call `update_compute_plan()`. The state will be reworked later.